src/main/webapp/placeGuides/ListPlaceGuideDisplayer.js

```diff
@@ -2,75 +2,132 @@
-  }
+class ListPlaceGuideDisplayer {
 
-  removeFromList(placeGuideId) {
+  static QUERY = {
+    "ALL_PUBLIC_IN_MAP_AREA": {
+      listTitle: "Discover Guides",
+      listSubTitle: "in selected map area"
+    },
+    "CREATED_ALL_IN_MAP_AREA": {
+      listTitle: "My Guides",
+      listSubTitle: "in selected map area"
+    },
+    "BOOKMARKED": {
+      listTitle: "Bookmarked Guides",
+      listSubTitle: ""    }
+  };
 
+  constructor(placeGuideDisplayQuery) {    
+    this._listPlaceGuideDisplayerDiv = document.getElementById("listPlaceGuideDisplayer");
+    this._listPlaceGuideDisplayerDiv.classList.add(
+        "list-group", "my-place-guide-list", "list-place-guide-displayer", "form-card");
+    this._placeGuideDisplayQuery = placeGuideDisplayQuery;
+    this._listPlaceGuideDisplayerDiv.appendChild(
+        this.createListTitle(
+            ListPlaceGuideDisplayer.QUERY[placeGuideDisplayQuery]));
   }
 
-  addToList(placeGuideId) {
+  get listPlaceGuideDisplayerDiv() {
+    return this._listPlaceGuideDisplayerDiv;
+  }
 
+  update(placeGuides) {
+    this.removeAllPlaceGuidesFromList();
+    this._listPlaceGuideDisplayerDiv.appendChild(
+        this.createListTitle(
+            ListPlaceGuideDisplayer.QUERY[this._placeGuideDisplayQuery]));
+    this.addPlaceGuidesToList(placeGuides);
   }
 
-  removePreviousPlaceGuidesFromList(placeGuides) {
-    for (const placeGuideId in this._placeGuidesOnList) {
-      if (this._placeGuidesOnList.hasOwnProperty(placeGuideId)) {
-        if (!placeGuides.hasOwnProperty(placeGuideId)) {
-          // This placeGuide is not needed anymore.
-          this.remove(placeGuideId);
-        }
-      }
-    }
+  removeAllPlaceGuidesFromList() {
+    while (this._listPlaceGuideDisplayerDiv.firstChild) {
+      this._listPlaceGuideDisplayerDiv.removeChild(
+          this._listPlaceGuideDisplayerDiv.lastChild);
+    };
   }
 
-  addNewPlaceGuidesToList(placeGuides) {
-    for (const placeGuideId in placeGuides) {
+  addPlaceGuidesToList(placeGuides) {
+    for (var placeGuideId in placeGuides) {
       if (placeGuides.hasOwnProperty(placeGuideId)) {
-        if (!this._placeGuidesOnList.hasOwnProperty(placeGuideId)) {
-          // new placeGuide should be constructed.
-          this._placeGuidesOnList[placeGuideId] =
-              this.
-                  constructPlaceGuideOnListFromPlaceGuide(
-                      placeGuides[placeGuideId]);
-          this.addToList(placeGuideId);
-        }
+        var constructedPlaceGuideOnListDiv = 
+            this.constructPlaceGuideOnListDivFromPlaceGuide(placeGuides[placeGuideId]);
+        this._listPlaceGuideDisplayerDiv.appendChild(constructedPlaceGuideOnListDiv);
       }
     }
   }
 
   remove(placeGuideId) {
-    // Delete from list.
-    this.removeFromList(placeGuideIde);
-    // Delete from memory.
-    delete this._placeGuidesOnList[placeGuideId];
+    const placeGuideDivId = "placeGuideOnList-" + "{" + placeGuideId + "}";
```

Comment:
I would suggest extracting the part for generating the div id in a separate function, assuming that we may need it multiple times. It's less error-prone.

Reply:
extracted a new function to accommodate getting the placeGuideOnListDivId.